calicoctl/commands/node/diags.go

```diff
@@ -155,7 +160,45 @@ such as transfer.sh using curl or similar.  For example:
 
 }
 
-// writeDiags executes the dignostic commans and outputs the result in the file
+// getHostedLogs will attempt to grab logs for any "calico" named containers for hosted installs.
+func getHostedLogs(logDir string) {
+	os.Mkdir(logDir, os.ModeDir)
+
+	// Get a list of Calico containers running on this Node.
+	var stderr bytes.Buffer
+	cmd := exec.Command("docker", "ps", "-a", "--filter", "\"name=calico\"", "--format", "{{.Names}}")
+	cmd.Stderr = &stderr
+	containers, err := cmd.Output()
+	if err != nil {
+		fmt.Printf("Could not run docker command: %s\n", stderr.String())
+		return
+	}
+
+	// No Calico containers found.
+	if string(containers) == "" {
+		log.Debug("Did not find any Calico containers")
+		return
+	}
+
+	fmt.Println("Copying logs from Calico containers")
+
+	// Grab the log for each container and write it as <containerName>.log.
```

Comment:
One thing we had talked about was also saving off the output of `docker ps` so that users could correlate containerName.log with a specific container / timestamp / etc.

i.e. if I'm looking at the directory this code generates, how do I know which of the 100 containers I'm interested in?

Reply:
I recall talking about it, but at the moment I'm having trouble seeing what this would buy the user, at the moment the only output of the docker command is a list of container names that have "calico" in them, which we use for grabbing the logs and using that name for the logfile.  I could run a similar query with more flags that could include some extra info, like start/creation time and save that out?

Reply:
The main thing we're looking for is the ability to correlate a log file with a specific iteration of a container.

So, being able to see the start/creation time is important.  Alternatively we can just encode that into the file name somehow.

Reply:
The names are already pretty long, I'll dump a file with the format:

```
{{.Names}}: {{.CreatedAt}}
```

Which will give us a mapping.